src/disco/containers/InstallButton.js

```diff
@@ -0,0 +1,63 @@
+import React, { PropTypes } from 'react';
+import { connect } from 'react-redux';
+
+import { gettext as _ } from 'core/utils';
+
+import 'disco/css/InstallButton.scss';
+import {
+  DOWNLOADING,
+  INSTALLED,
+  UNKNOWN,
+  validInstallStates as validStates,
+} from 'disco/constants';
+
+export class InstallButton extends React.Component {
+  static propTypes = {
+    handleClick: PropTypes.func,
+    handleChange: PropTypes.func,
+    guid: PropTypes.string,
+    progress: PropTypes.number,
+    slug: PropTypes.string.isRequired,
+    state: PropTypes.oneOf(validStates),
+  }
+
+  static defaultProps = {
+    state: UNKNOWN,
```

Comment:
I was thinking maybe `status` and `downloadProgress`. `installState` sounds better to me than `addonState` but it feels redundant to me. I'd prefer `status` but I'm not married to it.

I'm not sure this should be in component state since it has quite a few possible transitions and managing that all in the component seems out of scope. That's exactly what reducers are for. It's also just the `START_DOWNLOAD` and `START_UNINSTALL` actions I put in the reducer that are user generated, the rest happen from callbacks we passed into the AddonManager code.